Add ConfigProvider

geerteltink opened the pull request:
Adds ConfigProvider with basic configuration.

geerteltink committed:
Added ConfigProvider with basic configuration

geerteltink committed:
Bring composer.lock insync with composer.json

michalbundyra commented on composer.json:
```diff
@@ -51,6 +51,9 @@
             "dev-release-3.0.0": "3.0.x-dev"
+        },
+        "zf": {
+            "config-provider": "Zend\\Expressive\\Router\\ConfigProvider"
```
Again, I would consider changing namespace

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-zendrouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
only 2018 please for new files

webimpress commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-zendrouter/blob/master/LICENSE.md New BSD License
+ */
+
```
remove empty line before strict type declaration

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+    public function getDependencies() : array
+    {
+        return [
+            'factories' => [
```
`invokables` ?
https://github.com/zendframework/zend-expressive-skeleton/blob/release-3.0.0/src/ExpressiveInstaller/Resources/config/router-zend-router.php#L10

I would consider to have here `invokable` for `ZendRouter`, and `aliases` for `RouterInterface`.

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-zendrouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
2018 only for new files, please

webimpress commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-zendrouter/blob/master/LICENSE.md New BSD License
+ */
+
```
no new line before strict_type declaration

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+    public function testInvocationReturnsArray() : array
+    {
+        $config = ($this->provider)();
+        self::assertInternalType('array', $config);
```
`$this->`... ?

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+        self::assertInternalType('array', $config['dependencies']);
+        self::assertArrayHasKey('factories', $config['dependencies']);
+        self::assertInternalType('array', $config['dependencies']['factories']);
+        self::assertArrayHasKey(RouterInterface::class, $config['dependencies']['factories']);
```
`$this->` ... 5x ?

weierophinney commented:
@xtreamwayz Same comments from the fastroute package apply here:

- Use a subnamespace for the `ConfigProvider`.
- Use `$this->` instead of `self::` for now.

weierophinney committed:
Update to PSR-15

Updates dependencies to use 3.0.0alpha1 or later versions of
zend-expressive-router. Since these depend on PSR-15, tests that mocked
the http-interop `MiddlewareInterface` needed to be updated to PSR-15.

weierophinney committed:
Adds CHANGELOG entry for #35

weierophinney committed:
Merge branch 'feature/psr-15' into release-3.0.0

Close #35

weierophinney committed:
Merge pull request #34 from xtreamwayz/feature/config-provider

Add ConfigProvider

Conflicts:
	composer.lock

weierophinney committed:
Moves `ConfigProvider` into subnamespace

Moves the `ConfigProvider` into the subnamespace
`Zend\Expressive\Router\ZendRouter` in order to prevent naming
collisions.

weierophinney committed:
Updates copyright ranges

Reflects when files were added.

weierophinney committed:
Refactors ConfigProvider dependencies

- Aliases `RouterInterface` to `ZendRouter`
- Marks `ZendRouter` as an invokable

weierophinney committed:
Use consistent assertion style

Refactors `self::` to `$this->` to be consistent with other components.
We can change to `self::` usage later.

weierophinney committed:
Adds CHANGELOG entry for #34

weierophinney commented:
Thanks, @xtreamwayz!
